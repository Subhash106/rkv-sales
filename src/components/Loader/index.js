import React from 'react';
import { Backdrop } from '@mui/material';

export default function Loader() {
  return (
    <Backdrop sx={{ color: '#fff', zIndex: theme => theme.zIndex.drawer + 1 }} open={true}>
      <div>Loading...</div>
    </Backdrop>
  );
}
