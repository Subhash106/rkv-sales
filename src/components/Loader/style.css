.loader {
  position: relative;

  &__top,
  &__bottom {
    width: 6rem;
    height: 6rem;
    position: absolute;
    transform: translate(-50%, -50%);
    border-radius: 50%;
  }

  &__top {
    background: white;
    animation: animateTop 2s infinite;
  }

  &__bottom {
    background: white;
    animation: animateBottom 2s infinite;
  }
}

@keyframes animateBottom {
  0% {
    opacity: 1;
    width: 0;
    height: 0;
  }

  50% {
    opacity: 0.5;
    width: 3rem;
    height: 3rem;
  }

  100% {
    opacity: 0;
    width: 6rem;
    height: 6rem;
  }
}

@keyframes animateTop {
  0% {
    opacity: 0;
    width: 6rem;
    height: 6rem;
  }

  50% {
    opacity: 0.5;
    width: 3rem;
    height: 3rem;
  }

  100% {
    opacity: 1;
    width: 0rem;
    height: 0rem;
  }
}
