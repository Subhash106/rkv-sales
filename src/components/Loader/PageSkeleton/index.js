import { Skeleton } from '@mui/material';
import React from 'react';

import './style.css';

export default function PageSkeleton() {
  return (
    <div className="page-skeleton">
      <div className="p-xs">
        <Skeleton variant="rectangle" height="10vh" width="100%" animation="wave" />
      </div>
      <div className="p-xs">
        <Skeleton variant="circular" height="10vh" width="10vh" animation="wave" />
      </div>
      <div className="p-xs">
        <Skeleton variant="rounded" height="30vh" width="100%" animation="wave" />
      </div>
      <div className="p-xs">
        <Skeleton variant="rectangle" height="30vh" width="100%" animation="wave" />
      </div>
    </div>
  );
}
