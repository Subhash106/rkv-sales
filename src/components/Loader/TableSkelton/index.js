import { Skeleton } from '@mui/material';
import { number } from 'prop-types';
import React from 'react';

import './style.css';

export default function TableSkelton({ columns, rows }) {
  console.log('[...Array(rows).keys()]', [...Array(rows).keys()]);
  return (
    <table className="skeleton-table">
      <thead>
        <tr>
          {[...Array(columns).keys()].map(el => (
            <th key={el}>
              <Skeleton sx={{ fontSize: '2rem' }} animation="wave" />
            </th>
          ))}
        </tr>
      </thead>
      <tbody>
        {[...Array(rows).keys()].map(el => (
          <tr key={el}>
            {[...Array(columns).keys()].map(elR => (
              <td key={elR}>
                <Skeleton sx={{ fontSize: '2rem' }} animation="wave" />
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
}

TableSkelton.propTypes = {
  columns: number,
  rows: number
};

TableSkelton.defaultProp = {
  columns: 3,
  rows: 3
};
