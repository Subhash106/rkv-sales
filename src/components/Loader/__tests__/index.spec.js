import React from 'react';
import { shallow } from 'enzyme';

import Loader from '..';

describe('Loader test suite', () => {
  it('Should match snapshot', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
