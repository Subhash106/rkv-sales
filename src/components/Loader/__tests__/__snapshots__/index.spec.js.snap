// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader test suite Should match snapshot 1`] = `ShallowWrapper {}`;
